app/main.js

```diff
@@ -37,6 +37,16 @@ app.on('ready', () => {
     if (error) console.error('Failed to register protocol')
   })
 
+  // Download files
+  session.defaultSession.on('will-download', (event, item) => {
+    let location = dialog.showSaveDialog({ defaultPath: item.getFilename() })
+    if (location) {
+      item.setSavePath(location)
+    } else {
+      event.preventDefault()
```

Comment:
this is the 'Cancel' logic, isn't it? maybe add a comment to make it clear

Reply:
ok, done

Reply:
Thx!